code/game/machinery/requests_console.dm

```diff
@@ -137,6 +144,60 @@ var/list/obj/machinery/requests_console/requests_consoles = list()
 			if(!("[department]" in req_console_information))
 				req_console_information += department
 
+/obj/machinery/requests_console/proc/add_to_global_rc_list()
+	for(var/dept in master_department)
+		requests_consoles_categorised[dept] += src
+
+/obj/machinery/requests_console/proc/set_master_department(var/name)//this is fucking awful but less awful than updating all the maps and consoles
+	if(!isemptylist(master_department))
+		add_to_global_rc_list()
+		return "already setup"
+	var/list/command = list("Bridge","Captain's Desk","Chief Engineer's Desk","Chief Medical Officer's Desk","Head of Personnel's Desk","Head of Security's Desk","Research Director's Desk")
+	var/list/engineering = list("Atmospherics","Chief Engineer's Desk","Engineering","Mechanics")
+	var/list/medical = list("Chief Medical Officer's Desk","Genetics","Medbay","Chemistry","Virology")
+	var/list/research = list("Genetics","Research Director's Desk","Robotics","Science","Xenoarchaeology","Xenobiology","Mechanics")
+	var/list/security = list("Head of Security's Desk","Security")
+	var/list/service = list("Bar","Hydroponics","Kitchen","Head of Personnel's Desk")
+	var/list/cargo = list("Head of Personnel's Desk","Cargo Bay")
+	var/list/civillian = list("Pod Bay","Tool Storage","Chapel","EVA","Arrival shuttle","Locker Room","Janitorial","Head of Personnel's Desk")
+	for(var/subdept in command)
+		if(name == subdept)
+			master_department += "Command"
+			master_department_short = "c"
+	for(var/subdept in engineering)
+		if(name == subdept)
+			master_department += "Engineering"
+			master_department_short = "e"
+	for(var/subdept in medical)
+		if(name == subdept)
+			master_department += "Medical"
+			master_department_short = "m"
+	for(var/subdept in research)
+		if(name == subdept)
+			master_department += "Research"
+			master_department_short = "n"
+	for(var/subdept in security)
+		if(name == subdept)
+			master_department += "Security"
+			master_department_short = "s"
+	for(var/subdept in cargo)
+		if(name == subdept)
+			master_department += "Cargo"
+			master_department_short = "u"
+	for(var/subdept in service)
+		if(name == subdept)
+			master_department += "Service"
+			master_department_short = "d"
+	for(var/subdept in civillian)
+		if(name == subdept)
+			master_department += "Civillian"
```

Comment:
typo, also common seems more appropriate?

Reply:
yeah this one's my bad